WordPress/Classes/Networking/JetpackServiceRemote.m

```diff
@@ -59,4 +60,75 @@ - (void)validateJetpackUsername:(NSString *)username
+
+    [self.wordPressComRestApi GET:path
+                       parameters:nil
+                          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+                              NSDictionary *dict = (NSDictionary *)responseObject;
+                              BOOL isJetpack = [[dict numberForKey:@"jetpack"] boolValue];
+                              success(isJetpack, nil);
+
+                          } failure:^(NSError *error, NSHTTPURLResponse *httpResponse) {
+
+                              if ([error.domain isEqualToString:WordPressComRestApiErrorDomain]) {
+                                  NSDictionary *userInfo = error.userInfo;
+                                  NSString *errorKey = [userInfo stringForKey:WordPressComRestApi.ErrorKeyErrorCode];
+                                  BOOL isJetpack = NO;
+
+                                  // Jetpack is installed and connected but there is some other error.
+                                  // It could be a php error.  It could be that xmlrpc requests have been blocked
+                                  // via .htaccess :
+                                  // <Files xmlrpc.php>
+                                  // order allow,deny
+                                  // deny from all
+                                  // </Files>
+                                  // Or it could be due to some other unknown error that caues the site
+                                  // to return ah HTTP 500 status code when queried by the REST API.
+                                  // NOTE: IF XML-RPC is disabled via `add_filter( 'xmlrpc_enabled', '__return_false' );`
+                                  // the REST API currently still treates it as working (system.listMethods still operates).
+                                  if ([errorKey isEqualToString:@"jetpack_error"] && error.code == WordPressComRestApiErrorUnknown) {
+                                      // Site is inaccessible. (500 error)
+                                      isJetpack = YES;
+                                      error = [NSError errorWithDomain:JetpackServiceRemoteErrorDomain
+                                                                  code:JetpackServiceRemoteErrorSiteInaccessable
+                                                              userInfo:error.userInfo];
+
+                                  }
+
+                                  // The unauthorized key is a special case here.
+                                  // The API call does not require credentials and does not return a 403 itself.
+                                  // This combination occurs when Jetpack is installed but not connected to wpcom.
+                                  if ([errorKey isEqualToString:@"unauthorized"] && error.code == WordPressComRestApiErrorAuthorizationRequired) {
+                                      // Jetpack is disabled/installed but not connected. (403 error)
```

Comment:
This error will be present if Jetpack was ever activated and a connection initiated, even after deleting Jetpack.

For a new site, the API returns:

- Before Jetpack: 404 unknown_blog
- Install Jetpack: 404 unknown_blog
- Activate Jetpack: 404 unknown_blog
- Start connecting: 403 unauthorized
- Deactivate Jetpack: 403 unauthorized
- Delete Jetpack: : 403 unauthorized